lua-api-crates/serde-funcs/src/lib.rs

```diff
@@ -1,17 +1,67 @@
-use config::lua::get_or_create_module;
+use config::lua::{get_or_create_module, get_or_create_sub_module};
 use config::lua::mlua::{self, IntoLua, Lua, Value as LuaValue};
 use luahelper::lua_value_to_dynamic;
 use serde_json::{Map, Value as JValue};
 use std::collections::HashSet;
 use wezterm_dynamic::{FromDynamic, Value as DynValue};
+use serde_yaml;
+use toml;
+
 
 pub fn register(lua: &Lua) -> anyhow::Result<()> {
+    let serde_mod = get_or_create_sub_module(lua, "serde")?;
+
+    serde_mod.set("json_encode", lua.create_function(json_encode)?)?;
+    serde_mod.set("json_decode", lua.create_function(json_decode)?)?;
+    serde_mod.set("yaml_encode", lua.create_function(yaml_encode)?)?;
+    serde_mod.set("yaml_decode", lua.create_function(yaml_decode)?)?;
+    serde_mod.set("toml_encode", lua.create_function(toml_encode)?)?;
```

Comment:
and `toml_encode_pretty`.  I don't believe that the yaml crate offers a pretty version, otherwise I'd suggest that as well